### PR TITLE
fix(empty-state): #MAG-424 add folder provider for empty state

### PR DESCRIPTION
**Pull request by alicedraillard**
## Issue ticket number and link
[MAG-424](https://jira.support-ent.fr/browse/MAG-424)

## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [ ] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: frontend/src/providers/BoardsNavigationProvider/index.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import {
```
Comment (alicedraillard) [inline frontend/src/providers/BoardsNavigationProvider/index.tsx:1]: not used yet

File: frontend/src/providers/BoardsNavigationProvider/types.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { Dispatch, ReactNode, SetStateAction } from "react";
```
Comment (alicedraillard) [inline frontend/src/providers/BoardsNavigationProvider/types.ts:1]: not used yet

File: frontend/src/routes/app/index.tsx [hunk 3]
```diff
@@ -158,23 +160,25 @@ export const App = () => {
               modalProps={modalProps}
               onSetModalProps={setModalProps}
             />
-            <BoardList
+            {/* <BoardList
```
Comment (alicedraillard) [inline frontend/src/routes/app/index.tsx:163]: board list will be reactivated in next dev

File: frontend/src/components/folder-item/FolderItem.tsx [hunk 4]
```diff
@@ -15,9 +15,7 @@ import { UserRights } from "~/services/utils/share.utils";
 
 type FolderListProps = {
   folder: Folder;
-  folders: Folder[];
-  isSelected: boolean;
-  areFoldersLoading: boolean;
+  foldersData: Folder[];
```
Comment (ThomasSevagen) [inline frontend/src/components/folder-item/FolderItem.tsx:18]: ce state est dans le provider, je ne pense pas que le passage de prop soit utile

File: frontend/src/providers/FoldersNavigationProvider/useFoldersLogic.ts [hunk 5]
```diff
@@ -8,17 +8,19 @@ import { FOLDER_TYPE } from "~/core/enums/folder-type.enum";
 import { FolderTreeNavItem } from "~/models/folder-tree.model";
```
Comment (ThomasSevagen) [inline frontend/src/providers/FoldersNavigationProvider/useFoldersLogic.ts:8]: le fichier useFolderLogic peut être delete tout a été concentré dans l'index du provider

**Review by FlorentMr — APPROVED** [review]

**Review by ThomasSevagen — APPROVED** [review]

**Pull request merged**